eng/pipelines/templates/jobs/ci.yml

```diff
@@ -75,20 +75,19 @@ jobs:
           MatrixFilters: ${{ parameters.MatrixFilters }}
           MatrixReplace: ${{ parameters.MatrixReplace }}
           SparseCheckoutPaths: [ "sdk/", ".vscode"]
-          PreGenerationSteps:
-            - template: /eng/common/pipelines/templates/steps/save-package-properties.yml
-              parameters:
-                ServiceDirectory: ${{parameters.ServiceDirectory}}
+          ${{ if eq(parameters.ServiceDirectory, 'auto') }}:
+            EnablePRGeneration: true
+
+            PRMatrix:
+              Name: pr_test_base
+              Path: eng/pipelines/templates/stages/platform-matrix.json
```

Comment:
@scbedd I'm trying to think more about if we can avoid having to do the out of band yaml lookup, or at least solve the template conditional matrix config issue.

For the latter, I think what you can do is pass in all the configs (since we concat them a level up at https://github.com/Azure/azure-sdk-for-js/blob/f24289f5141277877ee26c212bc1c515b42a933d/eng/pipelines/templates/stages/archetype-sdk-client.yml#L61-L65), and then in the script just cross check the `MatrixConfigs` property on the package object with whether it actually exists in the `PRMatrix.Paths` object

```
PRMatrix:
  Paths:
  - ${{ each config in parameters.MatrixConfigs }}:
    - ${{ config.Path }}
```

For the former, is it possible to inspect the matrix config paths in the create-pr-job-matrix script, as passed by my above snippet, then parse the service directory out from each path and cross check it with service directory info that may be in the package properties? This would not work for any matrices that are referenced in a cross-service directory way (not that we do that anywhere).

So tl;dr

1. Only grab matrix config paths from the yaml as passed in by jobs/ci.yml (post-templating)
2. if possible, cross check config path service directories against package artifact name service directories
3. if #2 not possible, then where <pkg>.MatrixConfig paths are not in the PRMatrix.Paths list, exclude that matrix as it's been templated out in our pipeline context.

Reply:
> For the former, is it possible to inspect the matrix config paths in the create-pr-job-matrix script, as passed by my above snippet, then parse the service directory out from each path and cross check it with service directory info that may be in the package properties

I don't believe so. Remember that each Package-Property identifies it's associated `ci.yml` by checking the closest `ci.*.yml` and finding the artifactname aligning with the current package being scanned. This ALSO tells us where we could find the yml config (minus the compile time stuff(

> Only grab matrix config paths from the yaml as passed in by jobs/ci.yml

Remember that for `js - pullrequest`, we DONT have all those matrixconfigs (and additionalmatrixconfigs) because we're NOT coming directly through the `ci.yml`. We're coming up with the matrix _based on the set of changed files_.

Reply:
But, I'm picking up what you're screaming with passing the combined ConfigPaths as the default. that's smart and I'll do that

Reply:
Ah good point about the centralized top level yaml file, forgot about that key detail!